tests/test_integration.py

```diff
@@ -132,29 +134,6 @@ def test_integration(tmp_work_dir, docker_cleanup, requests_mock, monkeypatch):
     manifest = json.load(manifest_file.open())
     assert manifest["workspace"] == "testing"
     assert manifest["repo"] == str(repo_path)
-    assert set(manifest["actions"]) == set(
```

Comment:
I think we should now be asserting equivalent things about the state of the database, rather than just removing these assertions.

Reply:
Well now. I think I disagree. I think we should have tests that check that the system behaves correctly in terms of knowing what has been run and therefore correctly calculating what needs to be run. But the actual persistence is an implementation detail of that, isn't it?

Reply:
It was mainly the list of files produced that I was concerned with. There's now nothing that asserts that the jobs which ran produced the files we expected them to. (Checking the manifest wasn't _exactly_ asserting that either, but it was better than nothing.) Probably we should assert that each of those listed files now exists in the workspace.

I don't have any principled feelings here, I'm just aware that test coverage is already sub-optimal and I worry about lots of red assert statements.

Reply:
Oh well, if that assertion is missing altogether then I absolutely agree we need it. I'll have a look at adding a check for what's actually on disk.

Reply:
Done. Thank you.

Reply:
https://github.com/opensafely-core/job-runner/pull/279/commits/6c562fc3024d240db6b2ab10b9a8cd93ba0f8ad6

Reply:
https://github.com/opensafely-core/job-runner/pull/279/commits/20a17896012b698227c5615c7e89bf69fb71ebd5